### PR TITLE
add Ruby checker

### PR DESCRIPTION
**Pull request by JesterOrNot**

### CODE REVIEW

File: src/config-inferrer.spec.ts [hunk 1]
```diff
@@ -155,5 +155,27 @@ describe('config inferrer', () => {
...
+            "packages.config": ''
+        },{
+            tasks: [
+                {
+                    init: "nuget install"
+                }
+            ]
+        })
+    ),
+    it("[Ruby] Gemfile",
```
Comment (jankeromnes) [inline src/config-inferrer.spec.ts]: Nit: Local style is lowercase:

```suggestion
    it("[ruby] gemfile",
```

File: src/config-inferrer.spec.ts [hunk 2]
```diff
@@ -155,5 +155,27 @@ describe('config inferrer', () => {
                 }
             ]
         })
+    ),
+    it("[.NET] nuget",
```
Comment (jankeromnes) [inline src/config-inferrer.spec.ts]: Nit: Local style is lowercase:

```suggestion
    it("[dotnet] nuget",
```

File: src/config-inferrer.ts [hunk 3]
```diff
@@ -82,11 +90,16 @@ export class ConfigInferrer {
...
     }
 
-    protected async checkMake(ctx: Context) {
-        if (await ctx.exists('CMakeLists.txt')) {
-            this.addCommand(ctx.config, 'cmake .', 'init');
-        } else if (await this.isMake(ctx)) {
-            this.addCommand(ctx.config, 'make', 'init');
+    protected async checkStack(ctx: Context) {
+        if (await ctx.exists("stack.yml") || await ctx.exists("stack.yaml")) {
+            this.addCommand(ctx.config, "stack install", "init");
```
Comment (jankeromnes) [inline src/config-inferrer.ts]: I think you mean `stack.yaml`, not `stack.yml`, right?

https://docs.haskellstack.org/en/stable/stack_yaml_vs_cabal_package_file/

Comment (jankeromnes) [reply src/config-inferrer.ts]: Also, why do you prefer it over `package.yaml` or `*.cabal`?

Comment (jankeromnes) [reply src/config-inferrer.ts]: Also please add a test for this in the spec.

**Comment by jankeromnes** [comment]
I'm confused. So many different systems seem to use a `stack.yml` file.

File: src/config-inferrer.ts (hunk 3 above)
Comment (JesterOrNot) [reply src/config-inferrer.ts]: Wait `ctx.exists` supports globbing

**Comment by jankeromnes** [comment]
Rebased and cleaned up, same as https://github.com/gitpod-io/gitpod-yml-inferrer/pull/1#issuecomment-605130424

**Pull request merged**